lib/producer/events_producer.ex

```diff
@@ -5,14 +5,14 @@ defmodule Mississippi.Producer.EventsProducer do
 
   use GenServer
 
-  alias AMQP.Channel
+  alias Mississippi.Producer.EventsProducer.ExRabbitPoolConnection
   alias Mississippi.Producer.EventsProducer.Options
   alias Mississippi.Producer.EventsProducer.State
   require Logger
 
   # TODO should these be customizable?
   @connection_backoff 10000
-  @adapter ExRabbitPool.RabbitMQ
+  @connection unless Mix.env() == :test, do: ExRabbitPoolConnection, else: MockAMQPConnection
```

Comment:
Putting an `else` to `unless` is discouraged because it's a bit contrieved (see https://hexdocs.pm/elixir/case-cond-and-if.html#if-unless). Also, styler will probably be mad about this.
In #9 I changed it to
```elixir
   @connection if Mix.env() == :test, do: MockAMQPConnection, else: ExRabbitPoolConnection
  ```

Reply:
the whole thing was refactored to read values from the config. in any case, there is now styler in the branch so it would have """fixed""" that anyways